Generators/include/Generators/InteractionDiamondParam.h

```diff
@@ -0,0 +1,40 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \author R+Preghenella - October 2018
+
+#ifndef ALICEO2_EVENTGEN_INTERACTIONDIAMONDPARAM_H_
+#define ALICEO2_EVENTGEN_INTERACTIONDIAMONDPARAM_H_
+
+#include "SimConfig/ConfigurableParam.h"
+#include "SimConfig/ConfigurableParamHelper.h"
+
+namespace o2
+{
+namespace eventgen
+{
+
+/**
+   ** a parameter class/struct to keep the settings of
+   ** the interaction diamond (position and width) and 
+   ** allow the user to modify them 
+   **/
+
+struct InteractionDiamondParam : public o2::conf::ConfigurableParamHelper<InteractionDiamondParam> {
+  static constexpr int N = 3;
+  double Position[N] = { 0., 0., 0. }; //[N]
+  double Width[N] = { 0., 0., 0. };    //[N]
```

Comment:
For the arrays of fixed dimension you actually don't need to indicate the number of elements to store at ```//[N]``` (needed for pointers only)

Reply:
ah. Good to know. Thanks.

Reply:
In this case, we could get rid of the `N` completely, no?

Reply:
yes, one can use directly ```double Position[3] = { 0., 0., 0. };```

BTW, the mechanism of ```double* pointer; //[NtoStore]``` would not work anyway with ```constexpr NtoStore```, the size also should be a persistent (non static) member.

Reply:
`double position[3] = { 0., 0., 0. };` is implemented, with small p (also for `width`).
Also, the name of the key is now `Diamond` instead of `InteractionDiamond`.
Will update the description above.